searx/engines/deviantart.py

```diff
@@ -7,76 +7,70 @@
 categories = ['images']
 paging = True
 time_range_support = True
 
-# search-url
-base_url = 'https://www.deviantart.com/'
-search_url = base_url + 'search?page={page}&{query}'
-time_range_url = '&order={range}'
-
-time_range_dict = {'day': 11,
-                   'week': 14,
-                   'month': 15}
+time_range_dict = {
+    'day'   : 'popular-24-hours',
+    'week'  : 'popular-1-week',
+    'month' : 'popular-1-month',
+    'year'  : 'most-recent',
+}
 
+# search-url
+base_url = 'https://www.deviantart.com'
 
-# do search-request
 def request(query, params):
-    if params['time_range'] and params['time_range'] not in time_range_dict:
-        return params
 
-    params['url'] = search_url.format(page=params['pageno'],
-                                      query=urlencode({'q': query}))
+    # https://www.deviantart.com/search/deviations?page=5&q=foo
+
+    query =  {
+        'page' : params['pageno'],
+        'q'    : query,
+    }
     if params['time_range'] in time_range_dict:
-        params['url'] += time_range_url.format(range=time_range_dict[params['time_range']])
+        query['order'] = time_range_dict[params['time_range']]
 
-    return params
+    params['url'] = base_url + '/search/deviations?' + urlencode(query)
 
+    return params
 
-# get response from search-request
 def response(resp):
-    results = []
 
-    # return empty array if a redirection code is returned
-    if resp.status_code == 302:
-        return []
+    results = []
 
     dom = html.fromstring(resp.text)
 
-    # parse results
     for row in dom.xpath('//div[contains(@data-hook, "content_row")]'):
         for result in row.xpath('./div'):
-            link = result.xpath('.//a[@data-hook="deviation_link"]')[0]
-            url = link.attrib.get('href')
-            title = link.attrib.get('title')
-            thumbnail_src = result.xpath('.//img')[0].attrib.get('src')
-            img_src = thumbnail_src
-
-            # http to https, remove domain sharding
-            thumbnail_src = re.sub(r"https?://(th|fc)\d+.", "https://th01.", thumbnail_src)
-            thumbnail_src = re.sub(r"http://", "https://", thumbnail_src)
-
-            url = re.sub(r"http://(.*)\.deviantart\.com/", "https://\\1.deviantart.com/", url)
-
-            # append result
-            results.append({'url': url,
-                            'title': title,
-                            'img_src': img_src,
-                            'thumbnail_src': thumbnail_src,
-                            'template': 'images.html'})
-
-    # return results
+
+            a_tag = result.xpath('.//a[@data-hook="deviation_link"]')[0]
+            noscript_tag = a_tag.xpath('.//noscript')
+
+            if noscript_tag:
+                img_tag = noscript_tag[0].xpath('.//img')
+            else:
+                img_tag = a_tag.xpath('.//img')
+            if not img_tag:
+                continue
+            img_tag = img_tag[0]
+
+            results.append({
```

Comment:
Why have you changed the indentation here?

Reply:
This is a `[refactor]` patch. So I decided to switch from the [_"Aligned with opening delimiter"_](https://www.python.org/dev/peps/pep-0008/#indentation) style to the [ _"Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest"_](https://www.python.org/dev/peps/pep-0008/#indentation) style.

With an eye on the [Maximum Line Length](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) I've got used to add an extra level of indentation (the latter style).

IMO: The growing diff is a common drawback of the patch, when switching from one code style to another. But I think this additional diff is acceptable in a *rewrite/refactoring* patch.

Reply:
I am sorry if I was not clear. Personally I prefer this indentation, as if you add a new element to the end of the line, you do not have to add extra comma, thus decreasing the size of the diff. By changing the indentation I meant why did you put `:` at the same position in every line? 

If you add a new key which is longer than any of the existing keys, the whole dictionary has to be realigned, so the `:`s can be at the same vertical position again.

But I would be happy with this format:

```python
results.append({
    'template': 'images.html',
    'url': a_tag.attrib.get('href'),
    'img_src': img_tag.attrib.get('src'),
    'title': img_tag.attrib.get('alt'),
})
```

Reply:
> I meant why did you put : at the same position in every line?

It is a (uneasy) compromise I often use for readability reasons.   Its a weak example, but in long lines and more key/value pairs, it has its value:

````python
{
    'img_src': img_tag.attrib.get('src'),
    'title': img_tag.attrib.get('alt'),
}
````

IMO: aligned its more _eye catching_ that merely different attributes from the same element are used as values

````python
{
    'img_src': img_tag.attrib.get('src'),
    'title':   img_tag.attrib.get('alt'),
}
````

The big drawback of is ..

> If you add a new key which is longer 

Yes, hoping there is enough space left does not scale very good in the diff.

I rebased my branch, removed the alignment (also in the `time_range_dict`)  and force-pushed.